From: Alexander Block <ablock84@googlemail.com>
Date: Wed, 1 Aug 2012 12:46:05 +0200
Patch-mainline: 3.7
Git-commit: 3954096d4b1adab6fbb3c26de37ddd4f781e072f
References: FATE#312888
Subject: [PATCH] Btrfs: fix check for changed extent in
 is_extent_unchanged

The previous check was working fine, but this check should be
easier to read. Also, we could theoritically have some exotic
bugs with the previous checks.

Signed-off-by: Alexander Block <ablock84@googlemail.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -3818,8 +3818,8 @@ static int is_extent_unchanged(struct se
 		/*
 		 * Check if we have the same extent.
 		 */
-		if (left_disknr + left_offset_fixed !=
-				right_disknr + right_offset) {
+		if (left_disknr != right_disknr ||
+		    left_offset_fixed != right_offset) {
 			ret = 0;
 			goto out;
 		}
